adapters/httpapi_compact.c

```diff
@@ -1358,6 +1366,10 @@ HTTPAPI_RESULT HTTPAPI_CloneOption(const char* optionName, const void* value, co
     }
     else if (strcmp("TrustedCerts", optionName) == 0)
     {
+#ifdef DO_NOT_COPY_TRUSTED_CERTS_STRING
```

Comment:
Not saving certificate is interesting.  Let's take this offline tomorrow.

Reply:
By default I am copying the cert and saving it, which is current behavior. 

Note;Right now in base64 cert size is 6k (azure-c-sdk certs) and in MCU where these certs are flash not copying and simply passing pointer can help us reduce good amount of memory foot print